svtools/lmerge.py

```diff
@@ -240,280 +84,410 @@ def merge(BP, sample_order, v_id, use_product, include_genotypes=False):
+    ninefive_i_L_end = max_i_L
+    ninefive_i_L_total = p_L[max_i_L]
+    
+    while (ninefive_i_L_total < 0.95):
+        if (ninefive_i_L_start <= 0) and (ninefive_i_L_end >= (len(p_L)-1)):
+            break
+        ninefive_i_L_start = max(0, ninefive_i_L_start - 1)
+        ninefive_i_L_end = min(len(p_L)-1, ninefive_i_L_end +1)
+        ninefive_i_L_total = sum(p_L[ninefive_i_L_start:ninefive_i_L_end+1])
+
+    ninefive_i_L_start = ninefive_i_L_start - max_i_L
+    ninefive_i_L_end = ninefive_i_L_end - max_i_L
+
+    ninefive_i_R_start = max_i_R
+    ninefive_i_R_end = max_i_R
+    ninefive_i_R_total = p_R[max_i_R]
+    
+    while (ninefive_i_R_total < 0.95):
+        if (ninefive_i_R_start <= 0) and (ninefive_i_R_end >= len(p_R)-1):
+            break
+        ninefive_i_R_start = max(0, ninefive_i_R_start - 1)
+        ninefive_i_R_end = min(len(p_R)-1, ninefive_i_R_end +1)
+        ninefive_i_R_total = sum(p_R[ninefive_i_R_start:ninefive_i_R_end+1])
+
+    ninefive_i_R_end = ninefive_i_R_end - max_i_R
+    ninefive_i_R_start = ninefive_i_R_start - max_i_R
+    CIPOS95=str(ninefive_i_L_start) + ',' + str(ninefive_i_L_end)
+    CIEND95=str(ninefive_i_R_start) + ',' + str(ninefive_i_R_end)
+    return [CIPOS95, CIEND95]
+
+
+
+def combine_pdfs(BP, c, use_product, weight_by_support=False, weight_by_samples=False):
+  
+    L = []
+    R = []
+    for b_i in c:
+        b = BP[b_i]
+        L.append([b.start_l,b.end_l,b.p_l])
+        R.append([b.start_r,b.end_r,b.p_r])
+
+    [start_R, end_R, a_R] = l_bp.align_intervals(R)
+    [start_L, end_L, a_L] = l_bp.align_intervals(L)
+
+    p_L = [0] * len(a_L[0])
+    p_R = [0] * len(a_R[0])
+    wts = [1] * len(c)
+
+    for c_i in range(len(c)):
+
+        if weight_by_support:
+            #sys.stderr.write("support\n")
+            #sys.stderr.write(str(a_L[c_i])+"\n")
+            A = BP[c[c_i]].l.rstrip().split('\t', 10)
+            m = l_bp.to_map(A[7])
+            wt=int(m['SU'])
+            #sys.stderr.write(str(wt)+"\t"+m['SNAME']+"\n")
+            a_L[c_i]=[wt*ali for ali in a_L[c_i]]
+            #sys.stderr.write(str(a_L[c_i])+"\n")
+            a_R[c_i]=[wt*ari for ari in a_R[c_i]]
+        elif weight_by_samples:
+            #sys.exit(1)
+            #sys.stderr.write("samples\n")
+            A = BP[c[c_i]].l.rstrip().split('\t', 10)
+            m = l_bp.to_map(A[7])
+            wt=len(m['SNAME1'].split(','))
+            #sys.stderr.write(str(wt)+"\t"+m['SNAME']+"\n")
+            a_L[c_i]=[wt*ali for ali in a_L[c_i]]
+            a_R[c_i]=[wt*ari for ari in a_R[c_i]]      
 
-        for c_i in range(len(c)):
-            for i in range(len(a_L[c_i])):
-                p_L[i] += a_L[c_i][i]
+        for i in range(len(a_L[c_i])):
+            p_L[i] += a_L[c_i][i]
+
+        for i in range(len(a_R[c_i])):
+            p_R[i] += a_R[c_i][i]
 
-            for i in range(len(a_R[c_i])):
-                p_R[i] += a_R[c_i][i]
+    #sys.exit(1)
+    ALG = 'SUM'
```

Comment:
Would be nice to remove the commented out lines from this function before merging.